# Trait-based filtering, not co-evolution, structures microbiome assembly in marine sponges

Joëlle van der Sprong <sup>1,?</sup>; Vani Tirumalasetty <sup>1</sup>; Simone Schätzle <sup>1</sup>; Oliver Voigt <sup>1</sup>; Dirk Erpenbeck <sup>1,2</sup>; Gert Wörheide <sup>1,2,3</sup>; Sergio Vargas <sup>1,?</sup>

----------

<sup>1</sup> Department of Earth and Environmental Sciences, Palaeontology and Geobiology, Ludwig-Maximilians-Universit?t M?nchen, Munich, Germany

<sup>2</sup> GeoBio-Center, Ludwig-Maximilians-Universit?t M?nchen, Munich, Germany

<sup>3</sup> Bavarian State Collections of Palaeontology and Geology, Munich, Germany



<sup>?</sup> corresponding authors: j.vandersprong@uni-goettingen.de & s.vargas@lmu.de

----------

Sponge microbiomes are diverse and likely coevolve with their hosts. Here we analyse the microbiome of a large sample of haplosclerid sponges from the Red Sea. We use whole-genome clade specific markers to infer a robust phylogeny of these sponges and understand the evolution of the haplosclerid holo-microbiome.

The following code is based on the *dada2* tutorial available at [https://benjjneb.github.io/dada2/tutorial_1_8.html](https://benjjneb.github.io/dada2/tutorial_1_8.html)

and this other tutorial: [https://astrobiomike.github.io/amplicon/dada2_workflow_ex](https://astrobiomike.github.io/amplicon/dada2_workflow_ex)


## Abstract

Understanding how microbiomes evolve in concert with their hosts is crucial to addressing broader questions about biological individuality and the evolutionary dynamics of complex holobionts. Phylosymbiosis describes a pattern in which microbial community relationships parallel host phylogeny. While often interpreted as evidence for host–microbiome co-diversification, phylosymbiosis can also arise from ecological filtering through phylogenetically conserved host traits. Here, we test these alternative hypotheses in a taxonomically diverse group of marine sponges (Phylum Porifera, Order Haplosclerida) collected along the Arabian coastline in the Red Sea. We inferred a robust host phylogeny using 1,153 target-captured genome-wide loci from 144 specimens and characterised their prokaryotic microbiomes via V4 16S rRNA gene amplicon sequencing. Host clade identity explained roughly 50% of microbial community variation (PERMANOVA R² = 0.498, P = 0.001), demonstrating host-specific microbiome structuring. However, topological congruence analyses revealed high incongruence between host phylogeny and microbial dendrograms (nRF > 0.89), and phylogenetic distance was only weakly correlated with microbiome dissimilarity (Mantel r = 0.177). Partial Mantel tests confirmed that this weak signal was not affected by geography. Trait-based analyses revealed that high versus low microbial abundance (HMA–LMA) classification was a better predictor of microbiome composition than phylogenetic distance (Mantel r = 0.230 vs r = 0.177, respectively). Furthermore, phylogenetically distant clades with convergent HMA morphology harboured similar microbiome profiles but had unique ASV compositions, with between-trait dissimilarity exceeding within-trait dissimilarity. These findings indicate that sponge–microbiome associations arise primarily from ecological filtering through convergent functional traits, rather than host–microbe co-diversification.

## Description

insert description here...

### Citation



<a rel="license" href="http://creativecommons.org/licenses/by/4.0/"><img alt="Creative Commons License" style="border-width:0" src="https://i.creativecommons.org/l/by/4.0/88x31.png" /></a><br />This work is licensed under a <a rel="license" href="http://creativecommons.org/licenses/by/4.0/">Creative Commons Attribution 4.0 International License</a>.




