# Genome-wide exon capture uncovers host-microbiome co-diversification in Red Sea Sponges

Joëlle van der Sprong <sup>1,§</sup>; Vani Tirumalasetty <sup>1</sup>; Simone Schätzle <sup>1</sup>; Oliver Voigt <sup>1</sup>; Dirk Erpenbeck <sup>1,2</sup>; Gert Wörheide <sup>1,2,3</sup>; Sergio Vargas <sup>1,§</sup>

----------

<sup>1</sup> Department of Earth and Environmental Sciences, Palaeontology and Geobiology, Ludwig-Maximilians-Universität MÃ¼nchen, Munich, Germany

<sup>2</sup> GeoBio-Center, Ludwig-Maximilians-Universität München, Munich, Germany

<sup>3</sup> Bavarian State Collections of Palaeontology and Geology, Munich, Germany



<sup>§</sup> corresponding authors: j.vandersprong@uni-goettingen.de & s.vargas@lmu.de

----------

Sponge microbiomes are diverse and likely coevolve with their hosts. Here we analyse the microbiome of a large sample of haplosclerid sponges from the Red Sea. We use whole-genome clade specific markers to infer a robust phylogeny of these sponges and understand the evolution of the haplosclerid holo-microbiome.

The following code is based on the *dada2* tutorial available at [https://benjjneb.github.io/dada2/tutorial_1_8.html](https://benjjneb.github.io/dada2/tutorial_1_8.html)

and this other tutorial: [https://astrobiomike.github.io/amplicon/dada2_workflow_ex](https://astrobiomike.github.io/amplicon/dada2_workflow_ex)


## Abstract

insert abstract here...

## Description

insert description here...

### Citation



<a rel="license" href="http://creativecommons.org/licenses/by/4.0/"><img alt="Creative Commons License" style="border-width:0" src="https://i.creativecommons.org/l/by/4.0/88x31.png" /></a><br />This work is licensed under a <a rel="license" href="http://creativecommons.org/licenses/by/4.0/">Creative Commons Attribution 4.0 International License</a>.




