---
title: "Genome-wide exon capture uncovers host-microbiome co-diversification in Red Sea Sponges"
author: "Van der Sprong et al."
date: "`r Sys.Date()`"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

```{r prep}

if(!requireNamespace("dada2", quietly = TRUE)){
  print("dada2 not installed, installing now...")
  if(!requireNamespace("BiocManager", quietly = TRUE)) {
    install.packages("BiocManager")
    install_exit_status <- BiocManager::install(c("dada2", "phyloseq"), lib="/PATH/TO/FOLDER/R/x86_64-suse-linux-gnu-library/4.3")
  }else{
    install_exit_status <- BiocManager::install(c("dada2","phyloseq"), lib="/PATH/TO/FOLDER/R/x86_64-suse-linux-gnu-library/4.3")  
  }
}else{
  print("dada2 there, loading...")
  library(dada2)
  library(phyloseq)
  library(ggplot2)
  library(vegan)
  library(egg)
  library(ape)
  library(RColorBrewer)
  library(phytools)
}

```

## Load the clean reads

Samples "GW3237", "GW3325", "GW3404", "GW3416", "GW3494", "GW3580", "GW4166", "GW4251", "GW5875", "GW6059", "GW6104", "GW6140", "GW6158", and "GW6196" were discarded due to low read counts (<5000).

Samples "GW3524", "GW4232", "GW5899", and "GW6180" were discarded due to missing UCE data.



```{r load reads}

path <- "./Data/Red_Sea_Prokaryotic_Microbiomes/Reads/"
#list.files(path)

# Forward and reverse fastq filenames have format: SAMPLENAME_R1_001.fastq and SAMPLENAME_R2_001.fastq
fnFs <- sort(list.files(path, pattern="_R1_001.clean.fastq.gz", full.names = TRUE))
fnRs <- sort(list.files(path, pattern="_R2_001.clean.fastq.gz", full.names = TRUE))

# Extract sample names, assuming filenames have format: SAMPLENAME_XXX.fastq
sample.names <- sapply(strsplit(basename(fnFs), "_"), `[`, 1)

plotQualityProfile(fnFs[1:2])
plotQualityProfile(fnRs[1:2])

```


##  Place filtered files in filtered/ subdirectory
```{r filter reads}

filtFs <- file.path(path, "Dada2_Filtered", paste0(sample.names, "_F_dada2.clean.fastq.gz"))
filtRs <- file.path(path, "Dada2_Filtered", paste0(sample.names, "_R_dada2.clean.fastq.gz"))
names(filtFs) <- sample.names
names(filtRs) <- sample.names

filter_output <- filterAndTrim(fnFs, filtFs, fnRs, filtRs,
              maxN=0, maxEE=c(2,2), truncQ=2, rm.phix=TRUE,
              compress=TRUE, multithread=TRUE)

```

```{r learn errors}

errF <- learnErrors(filtFs, multithread=TRUE)
errR <- learnErrors(filtRs, multithread=TRUE)

plotErrors(errF, nominalQ=TRUE)

```

```{r Sample inference}

dadaFs <- dada(filtFs, err=errF, multithread=TRUE)
dadaRs <- dada(filtRs, err=errR, multithread=TRUE)

```

```{r merge reads}

merged_pairs <- mergePairs(dadaFs, filtFs, dadaRs, filtRs, verbose=TRUE)

```


```{r Get sequence table}

seqtab <- makeSequenceTable(merged_pairs)
seqtab.nochim <- removeBimeraDenovo(seqtab, method="consensus", multithread=TRUE, verbose=TRUE)
sum(seqtab.nochim)/sum(seqtab)

```
```{r track reads}

getN <- function(x) sum(getUniques(x))
track <- cbind(filter_output, sapply(dadaFs, getN), sapply(dadaRs, getN), sapply(merged_pairs, getN), rowSums(seqtab.nochim), rowSums(seqtab.nochim)/filter_output[,2])
# If processing a single sample, remove the sapply calls: e.g. replace sapply(dadaFs, getN) with getN(dadaFs)
colnames(track) <- c("input", "filtered", "denoisedF", "denoisedR", "merged", "nonchim", "surviving reads")
rownames(track) <- sample.names
#rownames(track[track[,1]<5000,])

```

```{r Assign taxonomy}

red_sea_taxa <- assignTaxonomy(seqtab.nochim,"./Data/Red_Sea_Prokaryotic_Microbiomes/SilvaDB/v138/silva_nr99_v138.1_train_set.fa.gz", multithread=TRUE)

red_sea_taxa <- addSpecies(red_sea_taxa, "./Data/Red_Sea_Prokaryotic_Microbiomes/SilvaDB/v138/silva_species_assignment_v138.1.fa.gz")

```

```{r Inspect taxonomy table}

red_sea_taxa.print <- red_sea_taxa # Removing sequence rownames for display only
rownames(red_sea_taxa.print) <- NULL
head(red_sea_taxa.print)


```

```{r extract ASV table, etc}

# giving our seq headers more manageable names (ASV_1, ASV_2...)
asv_seqs <- colnames(seqtab.nochim)
asv_headers <- vector(dim(seqtab.nochim)[2], mode="character")

for (i in 1:dim(seqtab.nochim)[2]) {
    asv_headers[i] <- paste(">ASV", i, sep="_")
}

# making and writing out a fasta of our final ASV seqs:
asv_fasta <- c(rbind(asv_headers, asv_seqs))
write(asv_fasta, "./Data/ASVs.fa")

# count table:
asv_tab <- t(seqtab.nochim)
row.names(asv_tab) <- sub(">", "", asv_headers)
write.table(asv_tab, "./Data/ASVs_Counts.tsv", sep="\t", quote=F, col.names=NA)

#taxonomy table:
row.names(red_sea_taxa.print) <- sub(">", "", asv_headers)
write.table(red_sea_taxa.print, "./Data/ASVs_Taxonomy.tsv", sep = "\t", quote=F, col.names=NA)


# #filter mitochondria, column 5 contains the family info
mt_ASVs <- sub(">", "", rownames(red_sea_taxa.print[which(red_sea_taxa.print[,5] == "Mitochondria"),]))
chloro_ASVs <- sub(">", "", rownames(red_sea_taxa.print[which(red_sea_taxa.print[,4] == "Chloroplast"),]))

asv_tab_noMt_noChloro <- asv_tab[!(row.names(asv_tab) %in% c(mt_ASVs, chloro_ASVs)),]

# count table:
write.table(asv_tab_noMt_noChloro, "./Data/ASVs_Counts_noMt_noChloro.tsv", sep="\t", quote=F, col.names=NA)

#taxonomy table:
red_sea_taxa.print_noMt_noChloro <- red_sea_taxa.print[!(row.names(red_sea_taxa.print) %in% c(mt_ASVs, chloro_ASVs)),]

write.table(red_sea_taxa.print_noMt_noChloro, "./Data/ASVs_Taxonomy_noMt_noChloro.tsv", sep = "\t", quote=F, col.names=NA)


```


The NMDS of the samples using groupings based on a phylogenetic analysis of the host sponges is presented below using the Bray-Curtis dissimilarity. 

```{r NMDS by groups}

#define a function to plot the nmds in ggplot2 using the same coordinates for each group of species
ggplot_nmds <- function(points, colors, xmax, xmin, ymax, ymin){

  #calculate centroids!
  nmds_group_centroids <- aggregate(cbind(MDS1,MDS2) ~ Clade, data = points, FUN = mean)

  #calculate lines!
  nmds_group_lines <- merge(setNames(nmds_group_centroids, c("Clade", "Cent.x", "Cent.y")), points[,c(1,2,4)], by='Clade', sort=FALSE)


  #all data have to have cols MDS1 and MDS2 to plot it in the same "canvas"
 nmds_plot <- ggplot(points, aes(MDS1,MDS2)) + 
                geom_point(aes(color = Clade), size = 5) + 
                geom_point(data=nmds_group_centroids, shape=21, size = 3, fill="white", color="black", stroke=1) +
                geom_segment(data=nmds_group_lines, mapping = aes(xend= Cent.x, yend=Cent.y)) +
                scale_color_manual(values = colors) + theme_bw() +
                xlim(xmin, xmax) + ylim(ymin, ymax) + 
    theme(
                  axis.text = element_text(size = 16),      # Ax labels
                  axis.title = element_text(size = 18),     # Ax title
                  legend.text = element_text(size = 22),    # Legend text
                  legend.title = element_text(size = 24)    # Legend title
                )

  #svg(filename = "../Plots/nmds_all_samples.svg")
  return(nmds_plot)
  #dev.off()
}


#calculate an NMDS with the samples
nmds <- metaMDS(t(asv_tab_noMt_noChloro), try=50000)

sample_info <- Sample_Information

nmds_points <- as.data.frame(nmds$points)
#check the samples are in the same order if needed
#rownames(nmds_points) == sample_info$Sample

#bind the nmds and sample info matrix
nmds_points <- cbind(nmds_points, sample_info)

##### VANAF HIER NMDS PLOTS AANPASSEN!!!!

#select only samples assigned to a clade.
nmds_points <- nmds_points[!is.na(nmds_points$Clade),]

example_clade_one <- c("G01","G02","G09","G10","G11","G12")
example_clade_one_col_vector <- c("blue3","cadetblue","darkorange1","magenta","gold1","cyan")


example_clade_one_plot <- ggplot_nmds(nmds_points[nmds_points$Clade %in% example_clade_one,], example_clade_one_col_vector, 0.8, -0.8, 0.8, -0.8)

ggsave("Example_Clade_One_Plot.svg", example_clade_one_plot, width = 30, height = 30, units = "cm")

###########

example_clade_two <- c("G18","G19","G20","G21","G22")
example_clade_two_col_vector <- c("tan1","darkseagreen","maroon2","black","red4")

example_clade_two_plot <- ggplot_nmds(nmds_points[nmds_points$Clade %in% example_clade_two,], example_clade_two_col_vector, 0.8, -0.8, 0.8, -0.8)

ggsave("Example_Clade_Two_Plot.svg", example_clade_two_plot, width = 30, height = 30, units = "cm")

#############

example_clade_three <- c("G13", "G14", "G15", "G16","G17")
example_clade_three_col_vector <- c("bisque2","seagreen1","midnightblue","lightcyan3","firebrick2")

example_clade_three_plot <- ggplot_nmds(nmds_points[nmds_points$Clade %in% example_clade_three,], example_clade_three_col_vector, 0.8, -0.8, 0.8, -0.8)

ggsave("Example_Clade_Three_Plot.svg", example_clade_three_plot, width = 30, height = 30, units = "cm")

############

example_clade_four <- c("G03", "G04", "G05", "G06","G07","G08")
example_clade_four_col_vector <- c("chocolate4","purple3","darkolivegreen","greenyellow","cornflowerblue","plum")

example_clade_four_plot <- ggplot_nmds(nmds_points[nmds_points$Clade %in% example_clade_four,], example_clade_four_col_vector, 0.8, -0.8, 0.8, -0.8)

ggsave("Example_Clade_Four_Plot.svg", example_clade_four_plot, width = 30, height = 30, units = "cm")

############

example_clade_five <- c("G01","G06","G12")
example_clade_five_col_vector <- c("blue3","greenyellow","cyan")

example_clade_five_plot <- ggplot_nmds(nmds_points[nmds_points$Clade %in% example_clade_five,], example_clade_five_col_vector, 0.8, -0.8, 0.8, -0.8)

ggsave("Example_Clade_Five_Plot.svg", example_clade_five_plot, width = 30, height = 30, units = "cm")

############

example_clade_six <- c("G03","G04","G07","G08","G09","G15","G16")
example_clade_six_col_vector <- c("chocolate4","purple3","cornflowerblue","plum","darkorange1","midnightblue","lightcyan3")

example_clade_six_plot <- ggplot_nmds(nmds_points[nmds_points$Clade %in% example_clade_six,], example_clade_six_col_vector, 0.8, -0.8, 0.8, -0.8)

ggsave("Example_Clade_Six_Plot.svg", example_clade_six_plot, width = 30, height = 30, units = "cm")


#Color palette, from: https://stackoverflow.com/questions/15282580/how-to-generate-a-number-of-most-distinctive-colors-in-r


```

I want to test for a significant correlation between the phylogenetic distance and the Bray-Curtis dissimilarity of the microbiomes.


```{r Phylo mantel}

#use APE to get a distance matrix from a phylogeny.

red_sea_phylo <- read.tree("./Data/RAxML_RSHaplos_25.tre")
red_sea_phylo <- drop.tip(red_sea_phylo, c("GW4249_2", "GW4246_2", "GW4286_2"))
  
red_sea_phylo_dist <- cophenetic(red_sea_phylo)
red_sea_phylo_dist[upper.tri(red_sea_phylo_dist)] <- ""
red_sea_micro_dist <- vegdist(t(asv_tab_noMt_noChloro), method = "bray")

row.names(red_sea_phylo_dist) %in% row.names(as.matrix(red_sea_micro_dist))

phylo_micro_mantel <- mantel(as.matrix(red_sea_micro_dist), red_sea_phylo_dist, method="kendall", permutations = 999, na.rm = T, parallel = 24)

```

```{r Anosim and Adonis2}


samples_with_no_clade_assignment <- sample_info[is.na(sample_info$Clade),]$Sample

dim(as.matrix(red_sea_micro_dist))
red_sea_micro_dist_noNAs <- as.matrix(red_sea_micro_dist)[!(rownames(as.matrix(red_sea_micro_dist)) %in% samples_with_no_clade_assignment), !(rownames(as.matrix(red_sea_micro_dist)) %in% samples_with_no_clade_assignment)]
dim(red_sea_micro_dist_noNAs)

rownames(as.matrix(red_sea_micro_dist)) == sample_info$Sample
rownames(as.matrix(red_sea_micro_dist_noNAs)) == sample_info[!(is.na(sample_info$Clade)),]$Sample

red_sea_clades_noNAs <- sample_info[!(is.na(sample_info$Clade)),]$Clade
table(red_sea_clades_noNAs)
anosim(red_sea_micro_dist_noNAs, red_sea_clades_noNAs, permutations = 999)

adonis2(red_sea_micro_dist_noNAs~red_sea_clades_noNAs)

#test post-hoc adonis.

#library(devtools)
#install_github("pmartinezarbizu/pairwiseAdonis/pairwiseAdonis")

red_sea_clades_noNAs_pairwise_adonis <- pairwiseAdonis::pairwise.adonis(x = red_sea_micro_dist_noNAs, factors = red_sea_clades_noNAs, p.adjust.m = "BH" ,perm = 999)

unique(red_sea_clades_noNAs)

```

```{r alpha diversity of the groups}

alpha <- fisher.alpha(t(asv_tab_noMt_noChloro))

#root_node <- getMRCA(red_sea_phylo, c("GW3092", "GW6015"))

#red_sea_phylo_rooted <- reroot(red_sea_phylo, root_node, position=red_sea_phylo$edge.length[which(red_sea_phylo$edge[,2]==root_node)]/2)

#write.tree(red_sea_phylo_rooted, file = "./Data/Phylogenies/Rooted_RAxML_RSHaplos_25.tre")

rooted_red_sea_phylo <- read.tree("./Data/Rooted_RAxML_RSHaplos_25.tre")

#find ancestral node to the clades
phylo_groups <- sample_info[,c(1,2)]
# rownames(phylo_groups) <- phylo_groups$Sample
# phylo_groups$Sample <- NULL

clade_names <- c()
clade_ancestral_nodes <- c()

for(i in unique(phylo_groups$Clade)){
  if(!is.na(i)) {
    selected_clade_samples <- phylo_groups[(phylo_groups$Clade == i & !is.na(phylo_groups$Clade)),1]
    clade_ancestral_nodes <- c(clade_ancestral_nodes, getMRCA(rooted_red_sea_phylo, selected_clade_samples))
    clade_names <- c(clade_names, i)
    }
}

#named vector of colors for the clades 
col_vector<-setNames(c("blue3", "cadetblue", "chocolate4","purple3",
              "darkolivegreen","greenyellow","cornflowerblue","plum",
              "darkorange1","magenta","gold1","cyan",
              "bisque2","seagreen1","midnightblue","lightcyan3",
              "firebrick2", "tan1","darkseagreen","maroon2",
              "black","red4"),
              unique(phylo_groups$Clade))

#match clades to colors
phylo_groups$Color <- col_vector[match(phylo_groups$Clade, names(col_vector))]

par(mfrow = c(1, 1))

plotTree.barplot(rooted_red_sea_phylo, alpha, ylim = c(0,Ntip(rooted_red_sea_phylo)), lwd=1, fsize = 0.25, ftype = "off", outline = F)

dotTree(rooted_red_sea_phylo, alpha, ylim = c(0,Ntip(rooted_red_sea_phylo)), lwd=1, fsize = 0.25, ftype = "off", outline = F)

#plot tree
plotTree(rooted_red_sea_phylo, ylim = c(0,Ntip(rooted_red_sea_phylo)), lwd=1, fsize = 0.25, ftype = "i", outline = F, xlim = c(0,2))

#add clade color bars to different
for(i in 1:Ntip(rooted_red_sea_phylo)){

   polygon_color <- phylo_groups[phylo_groups$Sample == rooted_red_sea_phylo$tip.label[i],3]
   polygon(x=c(0.75,0.80,0.80,0.75), y=i+c(-0.5, -0.5, 0.5, 0.5), col=polygon_color, border="transparent")
   polygon(x=c(0.85,0.90,0.90,0.85), y=i+c(-0.5, -0.5, 0.5, 0.5), col=polygon_color, border="transparent")

 }

#cladelabels(red_sea_phylo_rooted, clade_names, clade_ancestral_nodes, wing.length = 0, offset = 0.5)

```



<!-- ```{r Prep phyloseq object} -->

<!-- theme_set(theme_bw()) -->
<!-- samples.out <- rownames(seqtab.nochim) -->
<!-- #here load a df with sample to sample info for figure annotations, I guess -->
<!-- #samdf <- data.frame(Subject=subject, Gender=gender, Day=day) -->
<!-- #samdf$When <- "Early" -->
<!-- #samdf$When[samdf$Day>100] <- "Late" -->
<!-- #rownames(samdf) <- samples.out -->


<!-- ps <- phyloseq(otu_table(seqtab.nochim, taxa_are_rows=FALSE),  -->
<!--                #sample_data(samdf),# add sample data here. -->
<!--                tax_table(red_sea_taxa)) -->

<!-- plot_richness(ps, measures=c("Shannon", "Simpson"))#, x="Day", , color="When") -->


<!-- ps.prop <- transform_sample_counts(ps, function(otu) otu/sum(otu)) -->
<!-- ord.nmds.bray <- ordinate(ps.prop, method="NMDS", distance="bray", trymax=100) -->

<!-- plot_ordination(ps, ord.nmds.bray, title="Bray NMDS") -->

<!-- ``` -->


